core/json.go

```diff
@@ -64,6 +64,9 @@ func (p *JSON) UnmarshalJSON(bs []byte) error {
 	if p == nil {
 		return fmt.Errorf("cannot unmarshal into nil JSON")
 	}
+	if string(bs) == "null" {
+		return nil
+	}
```

Comment:
Why is this needed? This means we actually skip writing to `p`, which doesn't seem right.